tempfile_test.go

```diff
@@ -117,83 +117,172 @@ func TestOpenTempFileConflict(t *testing.T) {
+			wantPerm:    0o600,
 		},
 		{
-			name: "existing",
-			path: pathExisting,
+			name:        "tempfile existing",
+			path:        pathExisting,
+			useTempFile: true,
+			want:        "replaced:tempfile existing",
+			wantPerm:    0o600,
+		},
+		{
+			name:        "tempfile umask",
+			path:        filepath.Join(t.TempDir(), "masked"),
+			useTempFile: true,
+			umask:       0o377,
+			want:        "replaced:tempfile umask",
+			wantPerm:    0o600,
+		},
+		{
+			name:     "defaults",
+			path:     filepath.Join(t.TempDir(), "npf defaults"),
+			want:     "replaced:defaults",
+			wantPerm: 0o600,
+		},
+		{
+			name:     "fixed perm",
+			path:     filepath.Join(t.TempDir(), "npf new perm"),
+			options:  []Option{WithStaticPermissions(0o654)},
+			want:     "replaced:fixed perm",
+			wantPerm: 0o654,
+		},
+		{
+			name:     "umask perm 0644",
+			path:     filepath.Join(t.TempDir(), "npf umask perm"),
+			options:  []Option{WithPermissions(0o777)},
+			umask:    0o012,
+			want:     "replaced:umask perm 0644",
+			wantPerm: 0o765,
+		},
+		{
+			name:     "setup with perm",
+			path:     pathExistingWithPerm,
+			options:  []Option{WithStaticPermissions(0o754)},
+			want:     "replaced:setup with perm",
+			wantPerm: 0o754,
+		},
+		{
+			name:     "overwrite existing with perm",
+			path:     pathExistingWithPerm,
+			options:  []Option{WithExistingPermissions},
+			want:     "replaced:overwrite existing with perm",
+			wantPerm: 0o754,
+		},
+		{
+			name:     "use permissions from non-existing with unset",
+			path:     filepath.Join(t.TempDir(), "never before"),
+			options:  []Option{WithExistingPermissions},
+			want:     "replaced:use permissions from non-existing with unset",
+			wantPerm: 0o600,
+		},
+		{
+			name:     "use permissions from non-existing with mode",
+			path:     filepath.Join(t.TempDir(), "never before"),
+			options:  []Option{WithPermissions(0o633), WithExistingPermissions},
+			umask:    0o012,
+			want:     "replaced:use permissions from non-existing with mode",
+			wantPerm: 0o621,
+		},
+		{
+			name:     "use permissions from non-existing with fixed",
+			path:     filepath.Join(t.TempDir(), "never before"),
+			options:  []Option{WithExistingPermissions, WithStaticPermissions(0o612)},
+			want:     "replaced:use permissions from non-existing with fixed",
+			wantPerm: 0o612,
+		},
+		{
+			name:     "custom tempdir",
+			path:     filepath.Join(t.TempDir(), "foo"),
+			options:  []Option{WithTempDir(t.TempDir()), WithStaticPermissions(0o612)},
+			umask:    0o012,
+			want:     "replaced:custom tempdir",
+			wantPerm: 0o612,
 		},
 	} {
 		t.Run(tc.name, func(t *testing.T) {
-			pf, err := TempFile("", tc.path)
-			if err != nil {
-				t.Errorf("TempFile(%q) failed: %v", tc.path, err)
+			if tc.umask != 0 {
+				withUmask(t, tc.umask)
 			}
 
-			t.Cleanup(func() {
-				if err := pf.Cleanup(); err != nil {
-					t.Errorf("Cleanup() failed: %v", err)
+			t.Run("write", func(t *testing.T) {
```

Comment:
AFAICT, you’re using subtests to divide writing/checking.

However, subtests must be able to run independent of each other, which is not the case when structuring the code like this. E.g., `go test -run=TestPendingFileCreation/check` would fail, right?

Reply:
You're right. Merged the subtests back.